### PR TITLE
Clean up documentation for the pulse builder before release

### PR DESCRIPTION
**Pull request by taalexander**
### Summary
Clean up documentation for pulse builder before release.

### CODE REVIEW

**Comment by taalexander** [comment]
Just had a look and some pieces weren't rendering correctly. I'll update shortly.

File: qiskit/pulse/__init__.py [hunk 1]
```diff
@@ -145,9 +135,251 @@
...
+    ~qiskit.pulse.builder.measure_all
+    ~qiskit.pulse.builder.delay_qubits
+
+Circuit Gates
+-------------
+Use these circuit level gates within your pulse program.
+
+It is also possible to call a circuit with :func:`qiskit.pulse.builder.call`.
+
+..warning::
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:320]:
```suggestion
.. warning::
```

File: qiskit/pulse/__init__.py [hunk 2]
```diff
@@ -145,9 +135,251 @@
...
+    d0, d1, d2, d3, d4 = dc(0), dc(1), dc(2), dc(3), dc(4)
+
+    with pulse.build(name='Pulse Programming in') as pulse_prog:
+        pulse.play([1, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 1], d0)
+        pulse.play([1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0], d1)
+        pulse.play([1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0], d2)
+        pulse.play([1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0], d3)
+        pulse.play([1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0], d4)
+
+    style = SchedStyle(figsize=(8, 4))
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:162]: Shouldn't the figure size be reduced? 
image

File: qiskit/pulse/__init__.py [hunk 3]
```diff
@@ -145,9 +135,251 @@
...
+        pulse.set_frequency(5e9, d0)
+
+        with pulse.build() as temp_sched:
+            pulse.play(pulse.library.Gaussian(20, 1.0, 3.0), d0)
+            pulse.play(pulse.library.Gaussian(20, -1.0, 3.0), d0)
+
+        pulse.call(temp_sched)
+        pulse.acquire(30, a0, pulse.MemorySlot(0))
+
+    style = SchedStyle(figsize=(8, 4))
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:235]: Reduce figure size here too?
image

File: qiskit/pulse/__init__.py [hunk 4]
```diff
@@ -145,9 +135,251 @@
...
+    :toctree: ../stubs/
+
+    ~qiskit.pulse.builder.acquire_channel
+    ~qiskit.pulse.builder.control_channels
+    ~qiskit.pulse.builder.drive_channel
+    ~qiskit.pulse.builder.measure_channel
+
+Instructions
+------------
+Pulse instructions available within the builder interface.
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:207]: Maybe something like this?
```suggestion
Pulse instructions are available within the builder interface. Here's an example:
```

File: qiskit/pulse/__init__.py [hunk 5]
```diff
@@ -145,9 +135,251 @@
...
+    d1 = pulse.DriveChannel(1)
+
+    with pulse.build() as pulse_prog:
+        with pulse.align_right():
+            # this pulse will start at t=0
+            pulse.play(pulse.Constant(100, 1.0), d0)
+            # this pulse will start at t=80
+            pulse.play(pulse.Constant(20, 1.0), d1)
+
+    style = SchedStyle(figsize=(8, 4))
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:274]: Reduce figure size

File: qiskit/pulse/__init__.py [hunk 6]
```diff
@@ -145,9 +135,251 @@
...
+    ~qiskit.pulse.builder.circuit_scheduler_settings
+    ~qiskit.pulse.builder.frequency_offset
+    ~qiskit.pulse.builder.inline
+    ~qiskit.pulse.builder.pad
+    ~qiskit.pulse.builder.phase_offset
+    ~qiskit.pulse.builder.transpiler_settings
+
+Macros
+------
+Macros to help you add more complex functionality to your pulse
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:293]:
```suggestion
Macros help you add more complex functionality to your pulse
```

File: qiskit/pulse/__init__.py [hunk 7]
```diff
@@ -145,9 +135,251 @@
...
+
+    ~qiskit.pulse.builder.measure
+    ~qiskit.pulse.builder.measure_all
+    ~qiskit.pulse.builder.delay_qubits
+
+Circuit Gates
+-------------
+Use these circuit level gates within your pulse program.
+
+It is also possible to call a circuit with :func:`qiskit.pulse.builder.call`.
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:318]:
```suggestion
To use circuit level gates within your pulse program call a circuit
with :func:`qiskit.pulse.builder.call`.
```

File: qiskit/pulse/__init__.py [hunk 8]
```diff
@@ -145,9 +135,251 @@
...
+    ~qiskit.pulse.builder.cx
+    ~qiskit.pulse.builder.u1
+    ~qiskit.pulse.builder.u2
+    ~qiskit.pulse.builder.u3
+    ~qiskit.pulse.builder.x
+
+Utilities
+---------
+Use these utility functions to gather attributes about the backend and modify
+how your program is built.
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:349]:
```suggestion
The utility functions can be used to gather attributes about the backend and modify
how the program is built.
```

File: qiskit/pulse/builder.py [hunk 9]
```diff
@@ -25,6 +24,12 @@
 statements. For example, below we write a simple program that :func:`play`\s
 a pulse:
 
+from qiskit.visualization import SchedStyle
+
+style = SchedStyle(figsize=(20, 10))
+
+plt.rcParams['figure.figsize'] = [15, 15]
+
```
Comment (SooluThomas) [inline qiskit/pulse/builder.py:32]: Is this code supposed to be a code block?  You could add a double colon in L25 or
```suggestion
.. code-block:: python

    from qiskit.visualization import SchedStyle
    style = SchedStyle(figsize=(20, 10))
    plt.rcParams['figure.figsize'] = [15, 15]
```

File: qiskit/pulse/__init__.py [hunk 10]
```diff
@@ -116,6 +103,9 @@
    Schedule
    Instruction
 
+.. _pulse-builder:
```
Comment (lcapelluto) [inline qiskit/pulse/__init__.py:106]: What does this do?

File: qiskit/pulse/__init__.py [hunk 11]
```diff
@@ -145,9 +135,251 @@
 
    PulseError
 
-"""
+Pulse Builder (:mod:`~qiskit.pulse.builder`)
```
Comment (lcapelluto) [inline qiskit/pulse/__init__.py:138]: Not beta anymore?

File: qiskit/pulse/__init__.py [hunk 12]
```diff
@@ -145,9 +135,251 @@
...
+.. autosummary::
+    :toctree: ../stubs/
+
+    ~qiskit.pulse.builder.build
+
+Channels
+--------
+Methods to return the correct channels for the respective qubit indices.
+
+.. jupyter-execute::
```
Comment (lcapelluto) [inline qiskit/pulse/__init__.py:185]: Maybe next time stash the results rather than executing fresh every time

Comment (taalexander) [reply qiskit/pulse/__init__.py:185]: Is there a dramatic performance benefit to this? I imagine plots will go out of sync quickly if emitted as PNG.

File: qiskit/pulse/__init__.py (hunk 11 above)
Comment (taalexander) [reply qiskit/pulse/__init__.py:138]: See the warning below.

File: qiskit/pulse/__init__.py (hunk 10 above)
Comment (taalexander) [reply qiskit/pulse/__init__.py:106]: Unnecessary link, I've removed.

File: qiskit/pulse/__init__.py [hunk 13]
```diff
@@ -145,8 +137,254 @@
...
+    d1 = pulse.DriveChannel(1)
+
+    with pulse.build() as pulse_prog:
+        with pulse.align_right():
+            # this pulse will start at t=0
+            pulse.play(pulse.Constant(100, 1.0), d0)
+            # this pulse will start at t=80
+            pulse.play(pulse.Constant(20, 1.0), d1)
+
+    style = SchedStyle(figsize=(5, 2.5))
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:279]:
```suggestion
    style = SchedStyle(figsize=(3, 2), title_font_size=10, axis_font_size=8)
```
image

File: qiskit/pulse/__init__.py [hunk 14]
```diff
@@ -145,8 +137,254 @@
...
+        pulse.set_frequency(5e9, d0)
+
+        with pulse.build() as temp_sched:
+            pulse.play(pulse.library.Gaussian(20, 1.0, 3.0), d0)
+            pulse.play(pulse.library.Gaussian(20, -1.0, 3.0), d0)
+
+        pulse.call(temp_sched)
+        pulse.acquire(30, a0, pulse.MemorySlot(0))
+
+    style = SchedStyle(figsize=(5, 2.5))
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:239]:
```suggestion
    style = SchedStyle(figsize=(3, 2), title_font_size=10, axis_font_size=8, icon_font_size=10)
```
image

File: qiskit/pulse/__init__.py [hunk 15]
```diff
@@ -145,8 +137,254 @@
...
+    d0, d1, d2, d3, d4 = dc(0), dc(1), dc(2), dc(3), dc(4)
+
+    with pulse.build(name='pulse_programming_in') as pulse_prog:
+        pulse.play([1, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 1], d0)
+        pulse.play([1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0], d1)
+        pulse.play([1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0], d2)
+        pulse.play([1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0], d3)
+        pulse.play([1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0], d4)
+
+    style = SchedStyle(figsize=(5, 2.5))
```
Comment (SooluThomas) [inline qiskit/pulse/__init__.py:170]: Reducing figsize here:
```suggestion
    style = SchedStyle(figsize=(3, 2), title_font_size=10, axis_font_size=8)
```
image

Comment (taalexander) [reply qiskit/pulse/__init__.py]: Fixed in 0194a2960.

File: qiskit/pulse/__init__.py (hunk 14 above)
Comment (taalexander) [reply qiskit/pulse/__init__.py]: Fixed in 0194a2960.

File: qiskit/pulse/__init__.py (hunk 13 above)
Comment (taalexander) [reply qiskit/pulse/__init__.py]: Fixed in 0194a2960.

**Pull request merged**